bin/mail/2017-teacher-application-responses/teacher_application_decision_processor.rb

```diff
@@ -151,80 +151,57 @@ def load_workshop_map
-      @results[decision] << result
+    email_params.to_final_params.except(:decision).tap do |effective_params|
+      @results[decision] << effective_params
     end
   end
 
-  def teachercon?(workshop_string)
-    TEACHER_CONS.any? {|tc| workshop_string.include? tc}
-  end
-
   def process_accept(teacher_application, program, accepted_workshop, regional_partner_override)
+    # First, update the actual dashboard DB with this accepted workshop string.
+    save_accepted_workshop teacher_application, program, accepted_workshop, regional_partner_override
+
     # There are 2 kinds of acceptance, TeacherCon (ours) and Regional Partner.
-    # We can tell based on the accepted workshop
-    if teachercon? accepted_workshop
-      process_accept_teachercon teacher_application, program, accepted_workshop, regional_partner_override
+    if teacher_application.accepted_program.teachercon?
+      process_accept_teachercon teacher_application
     else
-      process_accept_partner teacher_application, program, accepted_workshop
+      process_accept_partner teacher_application
     end
   end
 
-  def process_accept_teachercon(teacher_application, program, accepted_workshop, regional_partner_override)
-    # First, update the actual dashboard DB with this accepted workshop string.
-    save_accepted_workshop teacher_application, program, accepted_workshop, regional_partner_override
-
+  def process_accept_teachercon(teacher_application)
     regional_partner_name = teacher_application.regional_partner_name
     raise "Missing regional partner name for application id: #{teacher_application.id}" if regional_partner_name.blank?
 
-    # TeacherCon string is in the format: 'dates : location'
-    dates, location = accepted_workshop.split(':').map(&:strip)
-    params = {
-      teachercon_location_s: location,
-      teachercon_dates_s: dates,
-      regional_partner_name_s: regional_partner_name
-    }
-    process :accept_teachercon, teacher_application, params
+    process :accept_teachercon, teacher_application
   end
 
-  def process_accept_partner(teacher_application, program, accepted_workshop)
+  def process_accept_partner(teacher_application)
     # First, make sure this is a valid workshop string (lookup_workshop will raise an error otherwise)
-    workshop_info = lookup_workshop accepted_workshop
-
-    # Next, update the actual dashboard DB with this accepted workshop string.
-    save_accepted_workshop teacher_application, program, accepted_workshop
+    workshop_info = lookup_workshop teacher_application.accepted_workshop
 
     params = {
       regional_partner_name_s: workshop_info[:partner_name],
       regional_partner_contact_person_s: workshop_info[:partner_contact],
       regional_partner_contact_person_email_s: workshop_info[:partner_email],
-      workshop_registration_url_s: "https://studio.code.org/pd/workshops/#{workshop_info[:id]}/enroll",
+      workshop_registration_url_s: workshop_info[:id],
```

Comment:
Per offline discussion, I'm going to change `transform` to output a new key and value. This will be in: `workshop_id_i`, out: `workshop_registration_url_s`